reroll-initiative/old/reroll-initiative.js

```diff
@@ -0,0 +1,260 @@
+
+        this._postUpdateCombatHook();
+    }
+
+    /**
+     * Define the default config for the module. Restricted to get only
+     */
+    static get DEFAULT_CONFIG() {
+        return {
+            reroll: true,
+            actorTypes: "all" //future feature
+        };
+    }
+
+    /**
+     * Define the settings metadata for the module. Restricted to get only
+     */
+    static get SETTINGS() {
+        return {
+            module: "reroll-initiative",
+            key: "rriSettings",
+            name: "Reroll-Initiative Settings",
+            hint: "Settings for Reroll-Initiative module",
+            default: RerollInitiative.DEFAULT_CONFIG,
+            type: Object,
+            scope: "world"
+        };
+    }
+
+    /**
+     * Register module settings with game settings
+     * @todo: Add any steps to occur when settings change
+     */
+    _registerSettings () {
+        game.settings.register(RerollInitiative.SETTINGS.module, RerollInitiative.SETTINGS.key, {
+            name: RerollInitiative.SETTINGS.name,
+            hint: RerollInitiative.SETTINGS.hint,
+            default: RerollInitiative.SETTINGS.default,
+            type: RerollInitiative.SETTINGS.type,
+            scope: RerollInitiative.SETTINGS.scope,
+            onChange: settings => {
+                
+                console.log("Module settings changed, new option values: ",settings)
+            }
+        });
+    }
+
+    /**
+     * Resets settings back to default
+     * @todo: maybe expand this to deregister and register settings
+     */
+    _applydefaultConfig() {
+        this.config = RerollIinitiative.DEFAULT_CONFIG;
+        console.log("Resetting reroll-initiative settings to defaults:",RerollInitiative.DEFAULT_CONFIG);
+        this._saveSettings();
+    }
+
+    /**
+     * Saves current class instance options back to game settings
+     */
+    async _saveSettings () {
+        await game.settings.set(RerollInitiative.SETTINGS.module,RerollInitiative.SETTINGS.key,this.config);
+    }
+
+    /**
+     * Loads current class instance settings from game settings
+     */
+    async _loadSettings () {
+        let config = await game.settings.get(RerollInitiative.SETTINGS.module,RerollInitiative.SETTINGS.key);
+        this.config = config;
+        console.log(this.config);
+    }
+    
+    /**
+     * Update a single module setting
+     * @param {String} setting The setting to change
+     * @param {*} value The new value of the option
+     */
+    static async updateSetting(setting,value){
+        let settings = game.settings.get(RerollInitiative.SETTINGS.module,RerollInitiative.SETTINGS.key);
+
+        if(settings.hasOwnProperty(setting)){
+            console.log(setting);
+            settings[setting] = value;
+            await game.settings.set(RerollInitiative.SETTINGS.module,RerollInitiative.SETTINGS.key,settings);
+        }
+        else{
+            console.error("Module setting : "+setting+" does not exist!");
```

Comment:
Is the pattern to console.error and not actually throw an error?

Reply:
It should probably be a try...catch block. Looking here: https://stackoverflow.com/questions/25377115/what-is-the-difference-between-throw-error-and-console-error I should probably confirm what behaviour is required.